File: apiconcierge/src/main/java/com/concierge/apiconcierge/models/budget/BudgetItem.java
package com.concierge.apiconcierge.models.budget;

import com.concierge.apiconcierge.models.budget.enums.StatusBudgetItemEnum;
import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SecondaryTable(name = "tb_company", pkJoinColumns = @PrimaryKeyJoinColumn(name = "id"))
@SecondaryTable(name = "tb_resale", pkJoinColumns = @PrimaryKeyJoinColumn(name = "id"))
@SecondaryTable(name = "tb_budget", pkJoinColumns = @PrimaryKeyJoinColumn(name = "id"))
@Entity
@Table(name = "tb_budget_item")
public class BudgetItem {

    @JoinColumn(table = "tb_company", referencedColumnName = "id")
    @Column(name = "company_id")
    private Integer companyId;

    @JoinColumn(table = "tb_resale", referencedColumnName = "id")
    @Column(name = "resale_id")
    private Integer resaleId;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @JoinColumn(table = "tb_budget", referencedColumnName = "id")
    @Column(name = "budget_id")
    private Integer budgetId;

    private StatusBudgetItemEnum status;

    private String code;

    private String description;

    private Integer quantity;

    private Double discount;

    private Double price;
}

File: system-concierge/src/app/models/budget/budget-item.ts
import { IBudgetItem } from "../../interfaces/budget/ibudget-item";

export class BudgetItem implements IBudgetItem{
    companyId: number = 0;
    resaleId: number = 0;
    id: string = '';
    status: string = '';
    budgetId: number = 0;
    ordem: number = 0;
    partId: number = 0;
    code: string = '';
    description: string = '';
    quantity: number = 0.0;
    discount: number = 0.0;
    price: number = 0.0;   
}